src/Microsoft.Azure.SignalR.Common/Utilities/AckHandler.cs

```diff
@@ -35,15 +40,27 @@ public Task<AckStatus> CreateSingleAck(out int id, TimeSpan? ackTimeout = defaul
             {
                 return Task.FromResult(AckStatus.Ok);
             }
-            var info = (IAckInfo<AckStatus>)_acks.GetOrAdd(id, _ => new SingleAckInfo(ackTimeout ?? _defaultAckTimeout));
-            if (info is MultiAckInfo)
+            var info = (IAckInfo<AckStatus>)_acks.GetOrAdd(id, _ => new SingleAckWithStatusInfo(ackTimeout ?? _defaultAckTimeout));
+            if (info is MultiAckWithStatusInfo)
             {
                 throw new InvalidOperationException();
             }
             cancellationToken.Register(() => info.Cancel());
             return info.Task;
         }
 
+        public Task<T> CreateSingleAck<T>(out int id, TimeSpan? ackTimeout = default, CancellationToken cancellationToken = default) where T : IMessagePackSerializable, new()
+        {
+            id = NextId();
+            if (_disposed)
+            {
+                return Task.FromResult(new T());
+            }
+            var info = (IAckInfo<IMessagePackSerializable>)_acks.GetOrAdd(id, _ => new SingleAckWithMessagePackPayloadInfo<T>(ackTimeout ?? _defaultAckTimeout));
+            cancellationToken.Register(info.Cancel);
```

Comment:
also validate `if (info is MultiAckWithStatusInfo)`?

Reply:
The `id` is a newly generated number, so the return `info` must be the one we added in the method.